domain/src/test/java/org/oppia/android/domain/exploration/ExplorationProgressControllerTest.kt

```diff
@@ -2372,16 +2502,29 @@ class ExplorationProgressControllerTest {
     submitMultipleChoiceAnswer(choiceIndex = 0)
     submitMultipleChoiceAnswer(choiceIndex = 0)
 
-    val eventLog = fakeAnalyticsEventLogger.getMostRecentEvent()
-    assertThat(eventLog).hasHintUnlockedContextThat {
+    val hintOfferedEvent = fakeAnalyticsEventLogger.getLoggedEvent {
+      it.context.activityContextCase == HINT_UNLOCKED_CONTEXT
+    }.also {
+      assert(it != null)
+    }
+    assertThat(hintOfferedEvent!!).hasHintUnlockedContextThat {
```

Comment:
I think you should use the same approach here as in the 
`testHint_lastHintWithNoSol_offeredThenViewed_logsViewHintEvt_logsProgressSavingSuccessEvt()` below to retrieve the list of the last 2 events, of which the `hintOfferedEvent` will be the [0], and the `progressSavingSuccessEvent` will be the [1].

Reply:
Done.

Reply:
The test is failing on robolectric after making the changes.

Reply:
Here is the stacktrace from CI:

```kotlin
org.oppia.android.domain.exploration.ExplorationProgressControllerTest > testHint_offered_logsHintOfferedEvent_logsProgressSavingSuccessEvent FAILED
    expected: HINT_UNLOCKED_CONTEXT
    but was : PROGRESS_SAVING_SUCCESS_CONTEXT       
```

From the test name, we want to assert that:
a) hint offerred event is logged
b) progress saving success event is logged

However, the test setup isn't done in such a way to allow this:

Can you try the following:

   ```
 val eventLogList = fakeAnalyticsEventLogger.getMostRecentEvents(2)

    val hintOfferedEvent = eventLogList[0])
    // use eventlogsubject to assert event details
   
 val progressSavingSuccessEvent = eventLogList[1])
 // use eventlogsubject to assert event details
```

Reply:
> Can you try the following:
> 
> ```
> val eventLogList = fakeAnalyticsEventLogger.getMostRecentEvents(2)
> 
>  val hintOfferedEvent = eventLogList[0])
>  // use eventlogsubject to assert event details
> 
> val progressSavingSuccessEvent = eventLogList[1])
> // use eventlogsubject to assert event details
> ```

I have already made those changes. PTAL at the code [here](https://github.com/oppia/oppia-android/blob/bdf303493987e84eef49c3e60b3e43dfa67bb58d/domain/src/test/java/org/oppia/android/domain/exploration/ExplorationProgressControllerTest.kt#L2447-L2455). The test is failing on Robolectric after making the suggested changes. Although, the test is passing locally.

I think test is failing maybe because the order of the events in which they are logged is changing. This was the reason why I had introduced `fun getLoggedEvent(predicate: (EventLog) -> Boolean): EventLog?` function.

Reply:
In this case I think it may be okay to add the getLoggedEvent functio.

Reply:
Done.